src/xmipp/libraries/reconstruction_adapt_cuda/angular_sph_alignment_gpu.cpp

```diff
@@ -0,0 +1,608 @@
+    ;
+}
+
+// usage ===================================================================
+void ProgAngularSphAlignmentGpu::defineParams()
+{
+    addUsageLine("Make a continuous angular assignment with deformations");
+	defaultComments["-i"].clear();
+	defaultComments["-i"].addComment("Metadata with initial alignment");
+	defaultComments["-o"].clear();
+	defaultComments["-o"].addComment("Metadata with the angular alignment and deformation parameters");
+    XmippMetadataProgram::defineParams();
+    addParamsLine("   --ref <volume>              : Reference volume");
+	addParamsLine("  [--mask <m=\"\">]            : Reference volume");
+	addParamsLine("  [--odir <outputDir=\".\">]   : Output directory");
+    addParamsLine("  [--max_shift <s=-1>]         : Maximum shift allowed in pixels");
+    addParamsLine("  [--max_angular_change <a=5>] : Maximum angular change allowed (in degrees)");
+    addParamsLine("  [--max_resolution <f=4>]     : Maximum resolution (A)");
+    addParamsLine("  [--sampling <Ts=1>]          : Sampling rate (A/pixel)");
+    addParamsLine("  [--Rmax <R=-1>]              : Maximum radius (px). -1=Half of volume size");
+    addParamsLine("  [--RDef <r=-1>]              : Maximum radius of the deformation (px). -1=Half of volume size");
+    addParamsLine("  [--l1 <l1=3>]                : Degree Zernike Polynomials=1,2,3,...");
+	addParamsLine("  [--l2 <l2=2>]                : Harmonical depth of the deformation=1,2,3,...");
+    addParamsLine("  [--optimizeAlignment]        : Optimize alignment");
+    addParamsLine("  [--optimizeDeformation]      : Optimize deformation");
+	addParamsLine("  [--optimizeDefocus]          : Optimize defocus");
+    addParamsLine("  [--phaseFlipped]             : Input images have been phase flipped");
+    addParamsLine("  [--regularization <l=0.01>]  : Regularization weight");
+	addParamsLine("  [--resume]                   : Resume processing");
+	addParamsLine("  [--fakeKernel]               : Uses fake kernel (CPU) instead of GPU kernel");
+	addParamsLine("  [--device <dev=0>]           : GPU device to use. 0th by default");
+    addExampleLine("A typical use is:",false);
+    addExampleLine("xmipp_angular_sph_alignment -i anglesFromContinuousAssignment.xmd --ref reference.vol -o assigned_anglesAndDeformations.xmd --optimizeAlignment --optimizeDeformation --depth 1");
+}
+
+// Produce side information ================================================
+void ProgAngularSphAlignmentGpu::createWorkFiles() {
+	auto gpu = GPU(device);
+	gpu.set();
+	// ? Could it be MetaDataVec (not MetaDataDb), dynamic_cast and no casting when equal pointer?
+	MetaDataVec *pmdIn = dynamic_cast<MetaDataVec*>(getInputMd());
+	MetaDataDb mdTodo, mdDone;
+	mdTodo =*pmdIn;
+	FileName fn(fnOutDir+"/sphDone.xmd");
+	if (fn.exists() && resume) {
+		mdDone.read(fn);
+		mdTodo.subtraction(mdDone, MDL_IMAGE);
+		mdTodo.write(fnOutDir + "/sphTodo.xmd");
+		mdTodo.read(fnOutDir + "/sphTodo.xmd");
```

Comment:
All this function is going to be refactored in the future version of the program. It will avoid this reading and make use of the metadata utilities to handle better this checkpoint functionality. However, it is still under testing, so I consider that it was better to not include it here. I would prefer to leave this function as it is for now and wait for the new version.